src/helper/selection-tools/bounding-box-tool.js

```diff
@@ -35,10 +35,12 @@ class BoundingBoxTool {
      * @param {Modes} mode Paint editor mode
      * @param {function} setSelectedItems Callback to set the set of selected items in the Redux state
      * @param {function} clearSelectedItems Callback to clear the set of selected items in the Redux state
+     * @param {function} setCursor Callback to set the visible mouse cursor
      * @param {!function} onUpdateImage A callback to call when the image visibly changes
      * @param {?function} switchToTextTool A callback to call to switch to the text tool
      */
-    constructor (mode, setSelectedItems, clearSelectedItems, onUpdateImage, switchToTextTool) {
+    constructor (mode, setSelectedItems, clearSelectedItems, setCursor, onUpdateImage, switchToTextTool) {
```

Comment:
Good point, thanks. The problem I faced with this is that I need to do `handleMouseMove` on those tools as well, i.e. I need to add [this](https://github.com/LLK/scratch-paint/pull/762/files#diff-a4da56a05d313254413ded8e7d51b386R144) to the other tools. But those tools don't have access to `getHitOptions`. I'll probably have to go with a solution like you mentioned [here](https://github.com/LLK/scratch-paint/pull/762/#discussion_r234003522) (passing that function around). Input on how exactly that should be done, i.e. where that `getHitOptions` function should be stored (so we aren't duplicating code everywhere)? ATM I think we just define the `method(val) { dispatch(method(val)) }`-style thing for reducers in every file that uses those reducers, but obviously that's not an option for this. Is there a util function somewhere for this type of function? (`getHitOptions` works well as a separate function, BTW, since it doesn't need to be tied to any class - it doesn't ever reference `this`.)

Reply:
(Resolved because code is understood, still have to make commits for oval/text/(bitmap) tools.)